errors.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+// type ErosError interface {
+//  Error() string
+//  ErrorCode() int
+// }
+
+type ErosError interface {
+    Error() string
+    Code() int
+}
+
+type ErosErrorImpl struct {
```

Comment:
This sounds very java-like. Could rename the interface ErosFailure and keep the ErosError name for the concrete implementation?